src/windows/line.rs

```diff
@@ -0,0 +1,150 @@
+                for i in 1..n_lines - 1 {
+                    // Valid because i + 1 < n_lines - 1 + 1
+                    let next_rva = self.lines.get_unchecked(i + 1).rva;
+                    let line = self.lines.get_unchecked_mut(i);
+                    line.len = next_rva - line.rva;
+                }
+
+                first_rva
+            } else {
+                // Valid because self.lines is not empty
+                self.lines.get_unchecked(0).rva
+            };
+
+            // Valid because self.lines is not empty
+            let last = self.lines.get_unchecked_mut(n_lines - 1);
+            last.len = sym_len - (last.rva - first_rva);
+        }
+    }
+
+    fn compute_rva(&mut self, address_map: &AddressMap) {
+        // the rva is the internal rva (needed to compute the length)
+        // So replace them (and eventually split lines) by the rva in the binary
+        let mut to_insert = Vec::new();
+
+        let mut is_sorted = true;
+        let mut last_rva = 0;
+
+        for (i, line) in self.lines.iter_mut().enumerate() {
+            let start = PdbInternalRva(line.rva);
+            let end = PdbInternalRva(line.rva + line.len);
+
+            // new_line won't be empty if and only if the range has been splitted
+            let mut new_line = Vec::new();
+
+            for (j, rg) in address_map.rva_ranges(start..end).enumerate() {
+                if j == 0 {
+                    line.rva = rg.start.0;
+                    line.len = rg.end - rg.start;
+                    is_sorted = is_sorted && last_rva <= line.rva;
+                    last_rva = line.rva;
+                } else {
+                    // The range has been splitted so need to insert the new mapping
+                    new_line.push(Line {
+                        rva: rg.start.0,
+                        num: line.num,
+                        len: rg.end - rg.start,
+                        file_id: line.file_id,
+                    });
+
+                    is_sorted = is_sorted && last_rva <= rg.start.0;
+                    last_rva = rg.start.0;
+                }
+            }
+
+            if !new_line.is_empty() {
+                to_insert.push((i + 1, new_line));
+            }
+        }
+
+        if !to_insert.is_empty() {
+            // Using extend_from_slice requires that the Line is cloneable
+            // Probably a slow path here but should be pretty rare
+
+            let mut new_lines = Vec::new();
+            let mut last = 0;
+            for (i, new_line) in to_insert.iter_mut() {
+                new_lines.extend_from_slice(&self.lines[last..*i]);
+                new_lines.append(new_line);
+                last = *i;
+            }
+
+            if last < self.lines.len() {
+                new_lines.extend_from_slice(&self.lines[last..]);
+            }
+            std::mem::replace(&mut self.lines, new_lines);
+        }
+
+        if !is_sorted {
+            self.lines.sort_by_key(|x| x.rva);
```

Comment:
Are we supposed to update `self.is_sorted` here?

Reply:
Done.
I moved the line stuff in the same place.
is_sorted is used to know if the lines are ordered by their rva (relative virtual address): this value is updated each time a line is added.
When we compute the line length we need to have ordered rva, so if they aren't then we sort them.